apps/aspd4525/fsw/unit_test/ut_aspd4525_custom_stubs.c

```diff
@@ -0,0 +1,153 @@
+/****************************************************************************
+*
+*   Copyright (c) 2017 Windhover Labs, L.L.C. All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions
+* are met:
+*
+* 1. Redistributions of source code must retain the above copyright
+*    notice, this list of conditions and the following disclaimer.
+* 2. Redistributions in binary form must reproduce the above copyright
+*    notice, this list of conditions and the following disclaimer in
+*    the documentation and/or other materials provided with the
+*    distribution.
+* 3. Neither the name Windhover Labs nor the names of its 
+*    contributors may be used to endorse or promote products derived 
+*    from this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+* LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+* ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+*****************************************************************************/
+
+#include "cfe.h"
+#include "utassert.h"
+#include "ut_aspd4525_custom_stubs.h"
+
+#include <string.h>
+
+Ut_ASPD4525_Custom_HookTable_t        Ut_ASPD4525_Custom_HookTable;
+Ut_ASPD4525_Custom_ReturnCodeTable_t  Ut_ASPD4525_Custom_ReturnCodeTable[UT_ASPD4525_CUSTOM_MAX_INDEX];
+
+
+void Ut_ASPD4525_Custom_Reset(void)
+{
+    memset(&Ut_ASPD4525_Custom_HookTable, 0x00, sizeof(Ut_ASPD4525_Custom_HookTable));
+    memset(&Ut_ASPD4525_Custom_ReturnCodeTable, 0x00, sizeof(Ut_ASPD4525_Custom_ReturnCodeTable));
+}
+
+
+void Ut_ASPD4525_Custom_SetFunctionHook(uint32 Index, void *FunPtr)
```

Comment:
Fantastic!  I don't remember showing you how to properly write custom hooks and stubs.  This is exactly how they SHOULD be written but rarely how they ARE written.  Good job!